streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java

```diff
@@ -156,6 +156,19 @@ public synchronized void put(final Bytes key, final byte[] value, final long tim
         cache.put(name, cacheFunction.cacheKey(keyBytes), entry);
     }
 
+    @Override
+    public byte[] fetch(final Bytes key, final long timestamp) {
+        validateStoreOpen();
+        final Bytes bytesKey = WindowStoreUtils.toBinaryKey(key, timestamp, 0, bytesSerdes);
+        final Bytes cacheKey = cacheFunction.cacheKey(bytesKey);
+        LRUCacheEntry entry = cache.get(name, cacheKey);
+        if (entry == null) {
```

Comment:
This is intentional: we did not put the value read from underlying back to cache since it may unnecessarily generate duplicates for the downstream, because of the caching eviction. Imagine cache size = 0, then each put here will result in a downstream duplicate.